data/tabular/train_test_split.py

```diff
@@ -1,209 +1,459 @@
-    # paths_to_data = filtered_paths
-
-    representations = []
-
-    for path in paths_to_data:
-        df = pd.read_csv(path)
-
-        if repr_col in df.columns and path not in TRACKED_DATASETS:
-            print("Processing", path.split("/")[0])
-            TRACKED_DATASETS.append(path)
-            representations.extend(df[repr_col].to_list())
-
-    repr_df = pd.DataFrame()
-    repr_df[repr_col] = list(set(representations))
-
-    if "SMILES" in repr_df.columns:
-        split = create_scaffold_split(
-            repr_df, seed=seed, frac=[train_size, val_size, test_size]
+    # make deterministic
+    np.random.seed(seed)
+    random.seed(seed)
+    dask_random_state = da.random.RandomState(seed)
+
+    def assign_split(
+        row,
+        test_smiles,
+        val_smiles,
+        train_frac,
+        test_frac,
+        random_state=dask_random_state,
+    ):
+        if row["SMILES"] in test_smiles:
+            return "test"
+        elif row["SMILES"] in val_smiles:
+            return "valid"
+        else:
+            # Assign random number for train/remaining split using the random state
+            random_value = random_state.random_sample()
+            if random_value < train_frac:
+                return "train"
+            elif random_value < train_frac + test_frac:
+                return "test"
+            else:
+                return "valid"
+
+    # we err toward doing more I/O but having simpler code to ensure we don't make anything stupid
+    all_yaml_files = get_yaml_files(data_dir)
+    smiles_yaml_files = [
+        file for file in all_yaml_files if has_yaml_file_smiles_column(file)
+    ]
+    # we filter those out that are in the to_scaffold_split list
+    not_scaffold_split_yaml_files = [
+        file for file in smiles_yaml_files if not is_in_scaffold_split_list(file)
+    ]
+
+    # if we debug, we only run split on the first 5 datasets
+    if debug:
+        not_scaffold_split_yaml_files = not_scaffold_split_yaml_files[:5]
+
+    # we run random splitting for each dataset but ensure that the validation and test sets
+    # do not contain any SMILES that are in the validation and test sets of the scaffold split
+
+    with open(val_smiles_path, "r") as f:
+        val_smiles = f.read().splitlines()
+
+    with open(test_smiles_path, "r") as f:
+        test_smiles = f.read().splitlines()
+
+    # some datasets are hundreds of GB, so we have to use dask
+    # we will first do a random split on the whole dataset
+    # then we flip the values based on whether a SMILES
+    # is in the validation or test set of the scaffold split
+    # we will then write the new data_clean.csv file
+
+    # if the is no data_clean.csv file, we run the transform.py script
+    # in the directory of the yaml file
+    # we will then read the data_clean.csv file and do the split
+    for file in not_scaffold_split_yaml_files:
+        print(f"Processing {file}")
+        run_transform(file)
```

Comment:
I wouldn't run that here, needs to be run before.

Reply:
I'm afraid we overrride, get a file we don't want, or this breaks because of some special dirs that have a different setup, e.g., bio ner, mol translation.